### PR TITLE
Fix #22908: Crash when passing through invalid wall door

### LINKED ISSUE #22908: [3df424] openrct2.exe: WallEntryGetDoorSound
<table>
    <tr>
        <td></td>
        <td>
            <b><a href="https://openrct2.sp.backtrace.io/p/openrct2/triage?_t=W3siZXJyb3IiOnsiZmluZ2VycHJpbnQiOiIzZGY0MjRlMTliODU5OWQ4OWFhZjQzZmJhNTFmMzg5Zjc2ODU0ZWVjZjA4ZDJjYjUyMDkzYWQzODM0YWM2NmU5IiwidHgiOjI3NjgxfSwicGx1Z2luIjp7InBsdWdpbklkIjoiZ2l0aHViIn0sImludGVncmF0aW9uIjp7ImlkIjo0NzJ9LCJpbnN0YW5jZSI6eyJhZGRyZXNzIjoiaHR0cHM6Ly9vcGVucmN0Mi5zcC5iYWNrdHJhY2UuaW8ifX1d&amp;fingerprint=3df424e19b8599d89aaf43fba51f389f76854eecf08d2cb52093ad3834ac66e9&amp;_tx=6c21&amp;time=quarter">New error: [openrct2]</a></b
            ><br /><b>1</b> occurrence since (Sat Oct 05 2024 15:18:14 GMT+0000 (Coordinated Universal Time))<br />
        </td>
    </tr>
</table>
<table>
    <tr>
        <td><strong>Application</strong></td>
        <td>openrct2.exe</td>
    </tr>
        <td><strong>Attributes</strong></td>
    <td>
                 <strong>application</strong> &mdash; 1        <ul>
                        <li>
                                <code>openrct2.exe</code> 
                                &mdash; 1 (100.00%)
            </li>
                     </ul>
                <strong>attachment_config.ini</strong> &mdash; 1        <ul>
                        <li>
                                <code>C:\Users\olivi\OneDrive\Documents\OpenRCT2\crash\18403ea5-89ff-4872-a4ec-8d9f5e56a7c4.ini B</code> 
                                &mdash; 1 (100.00%)
            </li>
                     </ul>
                <strong>assert_failure</strong> &mdash; 1        <ul>
                        <li>
                                <code>*</code> 
                                &mdash; 1 (100.00%)
            </li>
                     </ul>
                <strong>commit</strong> &mdash; 1        <ul>
                        <li>
                                b971d15 
                                &mdash; 1 (100.00%)
            </li>
                     </ul>
                <strong>fault.address</strong> &mdash; 1        <ul>
                        <li>
                                <code>0x0B</code> 
                                &mdash; 1 (100.00%)
            </li>
                     </ul>
                <strong>error.message</strong> &mdash; 1        <ul>
                        <li>
                                <code>EXCEPTION_ACCESS_VIOLATION_READ</code> 
                                &mdash; 1 (100.00%)
            </li>
                     </ul>
                <strong>version</strong> &mdash; 1        <ul>
                        <li>
                                <code>OpenRCT2, v0.4.14-13-gb971d15 (b971d15 on develop) provided by GitHub</code> 
                                &mdash; 1 (100.00%)
            </li>
                     </ul>
                <strong>_rxid</strong> &mdash; 1        <ul>
                        <li>
                                <code>31000000-4394-b30c-0000-000000000000</code> 
                                &mdash; 1 (100.00%)
            </li>
                     </ul>
                <strong>callstack</strong> &mdash; 1        <ul>
                        <li>
                                <code>{&#34;frame&#34;:[&#34;WallEntryGetDoorSound&#34;,&#34;play_scenery_door_close_sound&#34;,&#34;AnimateSceneryDoor&lt;0&gt;&#34;]}</code> 
                                &mdash; 1 (100.00%)
            </li>
                     </ul>
             </td>
         <tr>
        <td><strong>Classifiers</strong></td>
        <td>
                        <strong>invalid-read</strong><br />
                    </td>
    </tr>
         <tr>
        <td><strong>Frames</strong></td>
        <td>
            <code> WallEntryGetDoorSound play_scenery_door_close_sound AnimateSceneryDoor&lt;0&gt; </code>
        </td>
    </tr>
     </table>

### PR DESCRIPTION
**Pull request by Gymnasiast**
Previously, there was no check for `nullptr`. While adding one, I also tidied up the code around it. Hope it all makes sense.

### CODE REVIEW

File: src/openrct2/object/WallSceneryEntry.cpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "WallSceneryEntry.h"
+
+DoorSoundType WallSceneryEntry::getDoorSoundType() const
+{
+    return static_cast<DoorSoundType>((flags2 & WALL_SCENERY_2_DOOR_SOUND_MASK) >> WALL_SCENERY_2_DOOR_SOUND_SHIFT);
+}
```
Comment (tupaschoal) [inline src/openrct2/object/WallSceneryEntry.cpp:15]: Would it be better to return a `std::optional<DoorSoundType>`? Then we wouldn't need to have `null` entries on the vector.

Comment (ZehMatt) [reply src/openrct2/object/WallSceneryEntry.cpp:15]: I don't think std::optional makes sense here, its a basic enumeration, this is also a direct mapping from the object data.

File: src/openrct2/ride/Vehicle.cpp [hunk 2]
```diff
@@ -6246,14 +6248,17 @@ void Vehicle::UpdateAdditionalAnimation()
...
-    if (doorSoundType != 0)
+    if (wallEntry == nullptr)
+        return;
+
+    auto doorSoundType = wallEntry->getDoorSoundType();
+    if (doorSoundType == DoorSoundType::none)
+        return;
+
+    auto soundId = DoorOpenSoundIds[EnumValue(doorSoundType)];
+    if (soundId != OpenRCT2::Audio::SoundId::Null)
```
Comment (ZehMatt) [inline src/openrct2/ride/Vehicle.cpp:6259]: Is this check even required? We know that none maps to null and we check for none already.

Comment (Gymnasiast) [reply src/openrct2/ride/Vehicle.cpp:6259]: Strictly speaking, it isn’t required. Shall I remove it?

Comment (ZehMatt) [reply src/openrct2/ride/Vehicle.cpp:6259]: Its a condition that can never happen so I think its best to remove it.

**Review by ZehMatt — APPROVED** [review]

**Pull request merged**